### ISSUE TITLE
ROI selection tool broken in Slicer-4.13.0-2021-09-23 preview

### ISSUE THREAD

**Issue by masadcv** [Project-MONAI/MONAILabel#422]
We use ROI selection tool for selecting bounding box for scribbles method. It worked great in previous versions of Slicer (e.g. from two days ago in **Slicer-4.13.0-2021-09-21** it worked fine).

With the latest preview release (**Slicer-4.13.0-2021-09-23**), there seems to be two issues:
1. The ROI tool has been significantly changed. It is now under markup module I believe where I am having a difficult time initialising and marking up an `vtkMRMLAnnotationROINode` node (see screenshot of before and after update). It would be good to have some guidance/documentation on how we can do the same interaction that we did in the version from two days ago. If this already exists, can you please point me to it. 
2. The markup module crashes Slicer as soon as I click on ROI symbol from within the tool (see screenshot below)
![Screenshot from 2021-09-24 15-41-21](https://user-images.githubusercontent.com/3410225/134693679-34bdb267-1042-48e4-a8ad-b44dbad427ea.png)
In the terminal from where I run Slicer, it gives the following error messages:

> Switch to module:  "MONAILabel"
  Switch to module:  "Markups"
  GetItemByDataNode: Invalid data node to find
  GetItemByDataNode: Invalid data node to find
  error: [/home/user/Downloads/Slicer-4.13.0-2021-09-23-linux-amd64/bin/SlicerApp-real] exit abnormally - Report the problem.

## Screenshots
Before the update (version **Slicer-4.13.0-2021-09-21** )
![Screenshot from 2021-09-24 15-36-40](https://user-images.githubusercontent.com/3410225/134693003-1ecd93cf-1c82-4dbb-a0f1-2b1fe4bf70ff.png)

After the update (version **Slicer-4.13.0-2021-09-23**)
![Screenshot from 2021-09-24 15-38-58](https://user-images.githubusercontent.com/3410225/134693293-e4d61e34-c134-4252-9b2c-62df5c5c7c2f.png)

cc: @lassoan

**Comment by lassoan** [comment]
Thank you for the feedback and sorry for the disruption. In yesterday's big commit two major changes were made:
- the new markups toolbar was introduced, to clearly separate creation of new markups from placinc control points
- placement of _markups_ ROI nodes is emphasized over legacy _annotation_ ROI nodes

> The ROI tool has been significantly changed. It is now under markup module I believe where I am having a difficult time initialising and marking up an vtkMRMLAnnotationROINode node (see screenshot of before and after update). It would be good to have some guidance/documentation on how we can do the same interaction that we did in the version from two days ago. If this already exists, can you please point me to it.

It is probably a good time to switch to vtkMRMLMarkupsROINode. Markups ROI has lots of more visualization, interaction, and computation features, it can be rotated without transforms, etc. and it is designed to have the same or very similar API as the old annotation ROI node. So, hopefully the switch will not be to hard and the new features will compensate for this effort. As a short term solution, annotation ROI can still be used, it has not been removed, but it has to be placed using Annotations module.

I would also recommend to add a markups place widget to your module GUI so that you can set your ROI node active and start placement using a single button click. You cn set up a place widget like this (of course instead of show you would add it to your widget layout):

```python
roiNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsROINode")

placeWidget = slicer.qSlicerMarkupsPlaceWidget()
placeWidget.setButtonsVisible(False)
placeWidget.placeButton().show() 
placeWidget.placeMultipleMarkups = slicer.qSlicerMarkupsPlaceWidget.ForcePlaceSingleMarkup
placeWidget.setMRMLScene(slicer.mrmlScene)
placeWidget.setCurrentNode(roiNode)
placeWidget.show()
```

We will add buttons for resetting of the ROI and showing/hiding the ROI in the next few days, which can further simplify your module GUI.

> The markup module crashes Slicer as soon as I click on ROI symbol from within the tool (see screenshot below)

We've been testing these changes for several weeks, but probably programmatically you can set up a state that leads to this crash. Most likely the issue will go away if you switch to markups ROI, but of course this issue has to be fixed regardless. I could not reproduce the problem using the GUI, but I'll try to reproduce it now using MONAILabel.

**Comment by masadcv** [comment]
Many thanks @lassoan for your prompt response and for providing ways to incorporate `vtkMRMLMarkupsROINode` in our flow. As we already have a release ready in a day or two, it would be difficult to make the changes and test them before that. Therefore I will continue this work on after this release. 

> We've been testing these changes for several weeks, but probably programmatically you can set up a state that leads to this crash. Most likely the issue will go away if you switch to markups ROI, but of course this issue has to be fixed regardless. I could not reproduce the problem using the GUI, but I'll try to reproduce it now using MONAILabel.

You are right, indeed the error is there when MONAILabel extension was accessed before clicking on ROI. Here is how you can reproduce this: https://www.youtube.com/watch?v=wZyaaMNCqvs

**Comment by masadcv** [comment]
This may also be related to this. I cannot seem to make other methods that require points (deepgrow, deepedit) work with foreground clicks in the same updated slicer version. It always adds a background click even if foreground click button is selected. It works fine in previous version. 

Here is what is happening with this: https://www.youtube.com/watch?v=g08sBsVjbOY

**Comment by lassoan** [comment]
Thanks for the additional information. I'm struggling with some CUDA issues, hopefully I can make MONAILabel work and get all the issues fixed quickly. I'll keep you posted.

**Comment by lassoan** [comment]
I was able to reproduce the sticky place button issue. Until the problem is fixed, a workaround is to use the toolbar to switch between active markup nodes. A fix should come soon.

**Comment by lassoan** [comment]
I've pushed a [fix](https://github.com/Slicer/Slicer/commit/13a2b9f7cee41b7fd9667d429606a804f6ca874d) (just minutes before the nightly build cutoff time). This should fix the problem of the place widget not changing the active markup node. Please test and let me know if there is any problem.

Also let us know if have any request for changing/improving markups ROIs. For example, the markups ROI is placed by clicking in two corners (instead of clicking the center and a corner as it was done in annotation ROIs). There is another mode that creates the ROI as the bounding box of control points. We can add the center+corner click mode if someone strongly prefers that.

**Comment by masadcv** [comment]
Many thanks @lassoan , as soon as I get the new preview build - I will test and get back to you with confirmation.

I am also looking into the markup roi integration. Once we have this, we can test it out. I guess its just a matter of preference on how to give the interaction. Clicking two corners should work well for us, we will just need to show that to others in tutorial so they know what they are doing. One thing we found super useful in Slicer was adjustment of ROIs from within the 3D view.

**Comment by masadcv** [comment]
This maybe a minor comment but can we have a neutral default fill color for MarkupROI bounding box. Currently it is red color which maybe confusing for when we give scribbles (red scribbles are currently for background). 

I was able to manually change this for an already added node from Markups editor. Something like the following will work for a default color. 
![Screenshot from 2021-09-25 17-01-54](https://user-images.githubusercontent.com/3410225/134777894-0a75c283-b0ca-4c74-a826-019dc1ae49ae.png)

**Comment by lassoan** [comment]
Thanks for the useful feedback! The default markups display properties are stored in the application settings (users can change the default in Markups module by clicking "Set as default" button). Some users requested creating each ROI with a different color, so the default color may even vary in the future. So your module probably cannot assume a specific default color. 

If you have a preference for a specific ROI color then I would recommend to initialize the newly added ROI's color (and other display properties). You could also show the place widget's color selector, but it is probably more convenient for the user if you set the color automatically.

**Comment by masadcv** [comment]
> I've pushed a [fix](https://github.com/Slicer/Slicer/commit/13a2b9f7cee41b7fd9667d429606a804f6ca874d) (just minutes before the nightly build cutoff time). This should fix the problem of the place widget not changing the active markup node. Please test and let me know if there is any problem.

I can confirm that this is now fixed in the latest preview build (**Slicer-4.13.0-2021-09-26-linux-amd64**).

**Comment by lassoan** [comment]
Great! Thanks for testing.
